code/modules/surgery/organs/organ_external.dm

```diff
@@ -138,6 +138,30 @@
 	replaced(H)
 	return 1
 
+/obj/item/organ/external/attack(mob/living/carbon/C, mob/user)
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.species && H.species.name == "Machine" && src.status == ORGAN_ROBOT)
+			if(!H.get_organ(limb_name))
+				if(H == user)
+					H.visible_message("<span class='notice'>[H] is attempting to re-attach [src]...</span>")
+					do_mob(user, H, 120)
+					H.visible_message("<span class='warning'>[H] jams [src] into the empty socket!</span>",\
+					"<span class='notice'>You force [src] into your empty socket, and it locks into place!</span>")
+				else
+					H.visible_message("<span class='notice'>[H] is attempting to re-attach [src]...</span>")
+					do_mob(user, H, 60)
+					H.visible_message("<span class='warning'>[user] jams [src] into [H]'s empty socket!</span>",\
+					"<span class='notice'>[user] forces [src] into your empty socket, and it locks into place!</span>")
+				user.unEquip(src)
+				src.replaced(H)
```

Comment:
`src` is implied here also as you're calling `replaced()` on the object you're redefining the proc on.